aiocogeo/cog.py

```diff
@@ -323,3 +355,45 @@ def create_tile_matrix_set(self, identifier: str = None) -> Dict[str, Any]:
             "tileMatrix": list(reversed(matrices))
         }
         return tms
+
+
+
+@dataclass
+class CompositeReader(ReaderMixin):
+    readers: List[COGReader]
+
+    async def apply(self, func: Callable) -> List[Any]:
```

Comment:
Simply maps a function over all readers.  Used internally to implement the methods required by `ReaderMixin` but can also be used to do more complex logic (ex. ndvi).  `func` is any coroutine with signature `def func(reader: COGReader, *args, **kwargs)`